From a5b273f37a59054a19d09a478f4a85299644486b Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Sat, 9 Mar 2013 22:18:36 +0200
Subject: [PATCH 09/63] rm/ut: use m0_net_xprt_init()

Current implementation of m0_net_xprt_init() is empty, but we should
use it for uniformity.

rm_connect(): Add a comment explaining why two similar loops are needed.
---
 rm/ut/rcredits.c | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/rm/ut/rcredits.c b/rm/ut/rcredits.c
index b685828..3cdfa0e 100644
--- a/rm/ut/rcredits.c
+++ b/rm/ut/rcredits.c
@@ -181,6 +181,8 @@ static void rm_ctx_init(struct rm_context *rmctx)
 	struct m0_db_tx tx;
 
 	rmctx->rc_xprt = &m0_net_lnet_xprt;
+	rc = m0_net_xprt_init(rmctx->rc_xprt);
+	M0_UT_ASSERT(rc == 0);
 
 	rc = m0_net_domain_init(&rmctx->rc_net_dom, rmctx->rc_xprt,
 	                        &m0_addb_proc_ctx);
@@ -246,6 +248,7 @@ static void rm_ctx_fini(struct rm_context *rmctx)
 	m0_dbenv_fini(&rmctx->rc_dbenv);
 	m0_rpc_net_buffer_pool_cleanup(&rmctx->rc_bufpool);
 	m0_net_domain_fini(&rmctx->rc_net_dom);
+	m0_net_xprt_fini(rmctx->rc_xprt);
 }
 
 static void rm_connect(struct rm_context *src, const struct rm_context *dest)
@@ -645,12 +648,14 @@ static void remote_credits_utfini(void)
 	uint32_t i;
 
 	m0_rm_fop_fini();
-	for (i = 0; i < SERVER_NR; ++i) {
+
+	for (i = 0; i < SERVER_NR; ++i)
 		server_stop(i);
-	}
-	for (i = 0; i < SERVER_NR; ++i) {
+	/* Another loop is needed, because no rm_context should be finalised
+	 * until all servers are stopped. */
+	for (i = 0; i < SERVER_NR; ++i)
 		rm_ctx_fini(&rm_ctx[i]);
-	}
+
 	for (i = 0; i < TEST_NR; ++i) {
 		m0_clink_del_lock(&tests_clink[i]);
 		m0_clink_fini(&tests_clink[i]);
-- 
1.8.3.2

